domain/application/service/application.go

```diff
@@ -279,6 +279,13 @@ type ApplicationState interface {
 	// table.
 	NamespaceForWatchApplicationExposed() (string, string)
 
+	// NamespaceForWatchUnitForLegacyUniter returns the namespace identifiers
+	// for unit changes needed for the uniter. The first return value is the
+	// namespace for the unit's inherent properties, the second is the namespace
+	// of unit principals (used to watch for changes in subordinates), and the
+	// third is the namespace for the unit's resolved mode.
+	NamespaceForWatchUnitForLegacyUniter() (string, string, string)
```

Comment:
`NamespaceForLegacyWatchUnit` rolls a little better without for...for...

Reply:
It is a little cumbersome, but personally I would rather keep the standard format. This is all going to disappear eventually